src/server.h

```diff
@@ -119,6 +119,57 @@ typedef long long ustime_t; /* microsecond time type. */
 #define CONFIG_MIN_RESERVED_FDS 32
 #define CONFIG_DEFAULT_PROC_TITLE_TEMPLATE "{title} {listen-addr} {server-mode}"
 
+/* TODO:
+ * We can optimize this by ignoring the first N bits in the memory size, for
+ * example if we ignore the first 14 bits then we need 14 less buckets. This
+ * means we assume there are no clients with less than 2^14 (16K) memory usage,
+ * or at least we group them together with the up to 32K bucket.
```

Comment:
we should certainly improve. skipping the wasted first 14 buckets is a must. 
i think something simple, elegant and easy to understand is preferable.

Reply:
Added skipping of first 14bits and added new high buckets (2GB,4GB,+8GB). This seems good enough for me.
There's a still a comment in the code about the 2 layer bucketing idea, I'll archive it in this PR and remove it from the code once we feel ok with the current implementation.